tests/test_pcovr.py

```diff
@@ -431,6 +431,28 @@ def test_prefit_regressor(self):
         self.assertTrue(np.allclose(Yhat_regressor, Yhat_pcovr))
         self.assertTrue(np.allclose(W_regressor, W_pcovr))
 
+    def test_regressor_modifications(self):
+        regressor = Ridge(alpha=1e-8)
+        pcovr = self.model(mixing=0.5, regressor=regressor)
+
+        # PCovR regressor matches the original
+        self.assertTrue(regressor.get_params() == pcovr.regressor.get_params())
+
+        # PCovR regressor updates its parameters
+        # to match the original regressor
+        regressor.set_params(alpha=1e-6)
+        self.assertTrue(regressor.get_params() == pcovr.regressor.get_params())
+
+        # Fitting regressor outside PCovR fits the PCovR regressor
+        regressor.fit(self.X, self.Y)
+        self.assertTrue(hasattr(pcovr.regressor, "coef_"))
+
+        # PCovR regressor doesn't change after fitting
+        pcovr.fit(self.X, self.Y)
+        regressor.set_params(alpha=1e-4)
+        self.assertTrue(hasattr(pcovr.regressor_, "coef_"))
+        self.assertTrue(regressor.get_params() != pcovr.regressor_.get_params())
+
```

Comment:
Same here

Reply:
This I think should stay since it's different than checking the regressor shape. These tests are to ensure consistency between the regressor and KPCovR kernel parameters.

Reply:
Oops! I read this wrong -- my comment is more for lines 475-524

Reply:
Yes, I agree. If there already exists `sklearn` utilities that we could adopt to do these checks, then we could dramatically simplify these tests. We still might need _something_ to make the codecov bot happy, but the tests could be simplified.